protos/flyteidl/core/literals.proto

```diff
@@ -75,15 +75,80 @@ message Literal {
 
         // A map of strings to literals.
         LiteralMap map = 3;
+
+        // Supported Dense Vectors: Dense vectors are compact representation of slices or arrays of a univariate type
+        // Only some type are supported in dense vector representations. For other types, please use LiteralCollection
+
+        // DenseVector of type Int32
+        Int32Vector int32_vector = 4;
+        // DenseVector of type Int64
+        Int64Vector int_vector = 5;
+        // DenseVector of type float (32bit precision)
+        FloatVector float_vector = 6;
+        // DenseVector of type Double (64bit precision)
+        DoubleVector double_vector = 7;
+        // DenseVector of type Bool
+        BoolVector bool_vector = 8;
+        // DenseVector of type string
+        StringVector str_vector = 9;
     }
 }
```

Comment:
I think `Offloaded` should be added here instead... a Generic/String Scalar can be arbitrary large that we want it offloaded, same for Literal Collection/Map...

```suggestion
        StringVector str_vector = 9;
    }
    
    Offloaded offloaded = 10;
}
```

Reply:
ya I was thinking about that, the reason, why I did not add it here, is because we will then face major `TypeErasure` and lose the type information at runtime. I initially create a singular DenseVector which can be offloaded and one of the typed vectors. But, then if the value is offloaded at runtime, it is very hard to know the type. This has been causing some problems - i.e., we will always need to load the relevant interface.

In the approach I have taken, the shell is maintained, thus type is implicitly derivable

Reply:
If maintaining the type is a goal for offloading, we can add the LiteralType to the offload message...

If we follow the approach in this PR, how will we ever offload the literalcollection for example? It'll be impossible to while maintaining the type... Right?